packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -269,6 +257,8 @@ export const ComboBoxDataProviderMixin = (superClass) =>
       this._ensureItemsOrDataProvider(() => {
         this.dataProvider = oldDataProvider;
       });
+
+      this.clearCache();
```

Comment:
**note:** We don't want to get the `size` property reset by `_dataProviderFilterChanged` when setting a data provider. This is why I added a `clearCache()` call to the `_dataProviderChanged` observer instead of adding a `dataProvider` dependency to `_dataProviderFilterChanged`.

As an outcome, the `clearCache()` call in the `ready()` callback has become no longer necessary.